WMF Framework/Event Platform/StorageManager.swift

```diff
@@ -0,0 +1,185 @@
+import Foundation
+import CocoaLumberjackSwift
+
+class StorageManager: NSObject {
+
+    private var managedObjectContext: NSManagedObjectContext
+    private let pruningAge: TimeInterval = 60*60*24*30 // 30 days
+
+    static let shared: StorageManager? = {
+        let fileManager = FileManager.default
+        var storageDirectory = fileManager.wmf_containerURL().appendingPathComponent("Event Platform", isDirectory: true)
+
+        do {
+            try fileManager.createDirectory(at: storageDirectory, withIntermediateDirectories: true, attributes: nil)
+        } catch {
+            do {
+                let storageDirectoryIsReachable = try storageDirectory.checkResourceIsReachable()
+                if !storageDirectoryIsReachable {
```

Comment:
Hmm, I feel like there are more branches here than there is information or behavior. I assume this is the "always-try-to-create-and-fail-if-exists" pattern, maybe we can make it the "create-if-doesn't-exist" pattern instead? It would be nice if we didn't need nested do blocks.

```swift
do {
  if ! try storageDirectory.checkResourceIsReachable() {
    try fileManager.createDirectory(at: storageDirectory, withIntermediateDirectories: true, attributes: nil)
  }
} catch {
  DDLogError("EPC: Event Platform event storage directory is unreachable and cannot be created.")
  return nil
}
```

Reply:
Yeah, this code is a bit of a mess that I haven't looked back at since I got it working... 

Looking at @tonisevener's earlier implementation here (https://github.com/wikimedia/wikipedia-ios/pull/3648/commits/ab60d394ec4f866921af84f588158271e85e60ce#diff-ce0064ced62c208c2fdcccf81df261f4c1fa7aa8021e9e3d3a405a18a7495121R23-R27) it looks like we may be able to dispense even with the `checkResourceIsReachable` call and just call `createDirectory`; it'll return `true` if the directory already exists and `createIntermediates` is `true`.

https://developer.apple.com/documentation/foundation/filemanager/1407884-createdirectory
https://developer.apple.com/documentation/foundation/url/1779667-checkresourceisreachable

Reply:
Sounds good to me. No comment on the fact that a fn called `createDirectory` returns `true` if it fails to create a directory.

Reply:
Yeah, that earlier implementation was lifted straight from the old library (complete with the unused `var didGetDirectoryExistsError = false` var 😅). It seems to me like `createDirectory` could serve as both checks in one (I assume it would throw an error if it was unreachable, the documentation is a little hazy). At any rate I'm not aware of any problems we've had in the wild with the old library instantiation.